kuma/javascript/src/payments/subscription-form.jsx

```diff
@@ -60,17 +130,16 @@ export default function SubscriptionForm() {
                 </h2>
             </header>
             <form
-                ref={subscriptionFormRef}
                 name="subscription-form"
                 method="post"
-                onSubmit={submit}
+                onSubmit={openStripeModal}
                 disabled={isSubmitting}
             >
                 <label className="payment-opt-in">
                     <input
                         type="checkbox"
                         required="required"
-                        onClick={toggleButton}
+                        onClick={togglePaymentAuthorized}
```

Comment:
Perhaps, if you're in there fixing things, finally make this a controlled component. I don't trust uncontrolled checkboxes because they tend to get strangely remembered in browsers and the onClick will get out-of-sync.

Reply:
Good point, I feel the same way about uncontrolled components.